sql/src/main/java/io/crate/operation/PageResultListener.java

```diff
@@ -1,7 +1,37 @@
-package io.crate.operation;
+/*
+ * Licensed to Crate under one or more contributor license agreements.
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.  Crate licenses this file
+ * to you under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied.  See the License for the specific language governing
+ * permissions and limitations under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial
+ * agreement.
+ */
 
+package io.crate.operation;
 
+/**
+ * An interface to signal more paging data is needed.
+ * The PageResultListener has to take action (e.g. send a request to ask for more data from the source).
```

Comment:
What the PageResultListener does is not relevant, what's important is that it results in another `setBucket` or `setFailure` call (if `needMore=true`)

Reply:
Thanks, I've rephrased this to reflect your suggestion.